### PR TITLE
Added features

### PR DESCRIPTION
**Pull request by dhruv282**
- Added histogram role:
  The current deployment provides the histogram page under the stats URL and summary request. A new histogram role has been added with its appropriate functions in the backend.

- Added date range:
  This feature allows the user to view unique mementos in the summary page with in the selected date range.

- Added image selector:
  The user can now be able to view his/her selected mementos on the summary page. The grid tab in the summary page shows little close buttons on the upper right that allow users to select the thumbnails they would like to dismiss and can then click the update button to see the new selection which is updated in the slider and the GIF tabs as well. The user can revert back to the original selection by clicking the revert button.

### CODE REVIEW

**Comment by ibnesayeed** [comment]
@dhruv282 have you tested these changes and are you confident that these should be merged?

Also, currently this PR seems to have some conflicts, would you mind rebasing and resolving these conflicts then test properly before we can merge it?

Most of your commit messages are not really helpful for anyone who would look at the history of the repository. Please have a look at some guidelines on writing good commit messages (such as [this one](https://chris.beams.io/posts/git-commit/)) and exercise them in future.

File: public/scripts/gif.js [hunk 1]
```diff
@@ -2,9 +2,10 @@ var imageLinks = [];
 var stampedImages = [];
 
 function getImageArray(){
-
-	
+	console.log("Here's a gif!");
```
Comment (ibnesayeed) [inline public/scripts/gif.js]:
```suggestion
```

Do we need this debugging log here in the production code?

File: public/scripts/imageSliderScript.js [hunk 2]
```diff
@@ -55,6 +55,7 @@
 	});
 
 	function drawImageSlider(data){
+		console.log("I'm running!");
```
Comment (ibnesayeed) [inline public/scripts/imageSliderScript.js]:
```suggestion
```

Do we need this debugging log here in the production code?

File: public/scripts/gif.js (hunk 1 above)
Comment (dhruv282) [reply public/scripts/gif.js]: Sorry, we don't need that in the production code. I merged my code with [https://github.com/a-mabe/tmvis](url) to get the features Abigail added and this was part of it. I will get rid of those logs.

**Comment by dhruv282** [comment]
Sorry about that. I merged my code with [https://github.com/a-mabe/tmvis](url) to get the features Abigail added. I got some of the old commits with bad commit messages to my branch after merging our repositories. However, the new features we added before the merge explain the changes. Apologies for all this confusion.

**Comment by ibnesayeed** [comment]
This PR still has many conflicting files and cannot be merged. Please see me in my office.

**Comment by dhruv282** [comment]
So I was able to resolve the conflicts by rebasing and I also tested the rebased code. Everything is working as expected.

**Comment by ibnesayeed** [comment]
@dhruv282 it looks like you broke the live site by not describing necessary dependencies and not testing it in an isolated environment. Our docker deployment is now broken and giving the following error:

```
fs.js:27
const { Math, Object } = primordials;
                         ^
ReferenceError: primordials is not defined
    at fs.js:27:26
    at req_ (/app/node_modules/natives/index.js:143:24)
    at Object.req [as require] (/app/node_modules/natives/index.js:55:10)
    at Object.<anonymous> (/app/node_modules/webshot/node_modules/graceful-fs/fs.js:1:37)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:643:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:683:19)
    at require (internal/modules/cjs/helpers.js:16:16)
```

**Comment by dhruv282** [comment]
The error comes from the latest version of node, version 12. Switching from node: latest to node: 11.11.0 eliminates this error. The previous deployed version from April 8th this year was made before node v12 was released. I made necessary changes to the Dockerfile and tested it in an isolated environment. I'll make a new pull request with those changes.

**Comment by ibnesayeed** [comment]
@dhruv282 did you attempt to fix it for the latest version of node (potentially, in a way that works for both new and old versions of node)? If this can't work with the latest version of node, very soon there will be security issues discovered and we will be locked in with an unsupported node version.

**Comment by dhruv282** [comment]
I've started working towards that but using node v11 is a temporary fix to get the site back

**Pull request merged**